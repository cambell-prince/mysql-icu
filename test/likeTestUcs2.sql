-- Test using like for matching
set names utf8;

use mysql_icu_test;

drop table if exists likeTest;
create table likeTest  (
    id INT NOT NULL PRIMARY KEY AUTO_INCREMENT,
    phrase VARCHAR(128)
) collate ucs2_icu_my_MM_ci;

insert into likeTest (phrase) values 
('မြန်မာစကား'),
('စကာလုံးများ'),
('ဘယ်နှလုံးလဲ။'),
('ဘာလုပ်မလဲ။'),
('မြန်မာဘာသာစကား'),
('မြန်မာလူမျို'),
('မြန်မာနိုင်ငံ'),
('နိုင်ငံရေး'),
('လူများတယ်'),
('အင်္ဂလိန်ကမြန်မာပြည်ကိုလားတယ်။');

select * from likeTest where phrase like '%';
select 'မ%';
select * from likeTest where phrase like 'မ%';
select 'မြ%';
select * from likeTest where phrase like 'မြ%';
select '%န%';
select * from likeTest where phrase like '%န%';
select '%န်%';
select * from likeTest where phrase like '%န်%';
select '%လုံး%';
select * from likeTest where phrase like '%လုံး%';
select '%မျ%';
select * from likeTest where phrase like '%မျ%';
select  '%များ';
select * from likeTest where phrase like '%များ';
select  '%မျ%';
select * from likeTest where phrase like '%မျ%';
select  '%မာ%။';
select * from likeTest where phrase like '%မာ%။';
select  '%မာ%ကား';
select * from likeTest where phrase like '%မာ%ကား';
select  '%မာ%ကို%';
select * from likeTest where phrase like '%မာ%ကို%';
select  'မြန်%မာ%';
select * from likeTest where phrase like 'မြန်%မာ%';
select  '%က__';
select * from likeTest where phrase like '%က__';
select  '_ူ';
select * from likeTest where phrase like '_ူ';
select  '%မာ_ကား%';
select * from likeTest where phrase like '%မာ_ကား%';





