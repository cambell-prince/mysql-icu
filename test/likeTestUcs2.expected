id	phrase
1	မြန်မာစကား
2	စကာလုံးများ
3	ဘယ်နှလုံးလဲ။
4	ဘာလုပ်မလဲ။
5	မြန်မာဘာသာစကား
6	မြန်မာလူမျို
7	မြန်မာနိုင်ငံ
8	နိုင်ငံရေး
9	လူများတယ်
10	အင်္ဂလိန်ကမြန်မာပြည်ကိုလားတယ်။
မ%
မ%
မြ%
မြ%
id	phrase
1	မြန်မာစကား
5	မြန်မာဘာသာစကား
6	မြန်မာလူမျို
7	မြန်မာနိုင်ငံ
%န%
%န%
%န်%
%န်%
id	phrase
1	မြန်မာစကား
5	မြန်မာဘာသာစကား
6	မြန်မာလူမျို
7	မြန်မာနိုင်ငံ
10	အင်္ဂလိန်ကမြန်မာပြည်ကိုလားတယ်။
%လုံး%
%လုံး%
id	phrase
2	စကာလုံးများ
3	ဘယ်နှလုံးလဲ။
%မျ%
%မျ%
%များ
%များ
id	phrase
2	စကာလုံးများ
%မျ%
%မျ%
%မာ%။
%မာ%။
id	phrase
10	အင်္ဂလိန်ကမြန်မာပြည်ကိုလားတယ်။
%မာ%ကား
%မာ%ကား
id	phrase
1	မြန်မာစကား
5	မြန်မာဘာသာစကား
%မာ%ကို%
%မာ%ကို%
id	phrase
10	အင်္ဂလိန်ကမြန်မာပြည်ကိုလားတယ်။
မြန်%မာ%
မြန်%မာ%
id	phrase
1	မြန်မာစကား
5	မြန်မာဘာသာစကား
6	မြန်မာလူမျို
7	မြန်မာနိုင်ငံ
%က__
%က__
_ူ
_ူ
%မာ_ကား%
%မာ_ကား%
id	phrase
1	မြန်မာစကား
